### PR TITLE
Master cpp

### PR DESCRIPTION
**Pull request by SauravDharwadkar**
added folder conversion , fixed latest issue

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -15,3 +15,5 @@ add_executable(
         zstr/fstream.h zstr/zstr.h
 )
 target_link_libraries(tgs_to_gif ${CONAN_LIBS} rlottie)
+target_compile_options(tgs_to_gif 
+                       PUBLIC -fconcepts)
```
[the remark this answers is not in the document]
Comment (SauravDharwadkar) [reply CMakeLists.txt:19]: i passed auto as auto as argument and compiler gives waning so i added this flag to hide warnings

[the remark this answers is not in the document]
Comment (SauravDharwadkar) [reply CMakeLists.txt:19]: tgs-to-gif/main.cpp:18:86: warning: use of ‘auto’ in parameter declaration only available with ‘-fconcepts’
   18 |  filePath , auto width , auto height , auto alphaThreshold , auto output , auto fps){

**Comment by SauravDharwadkar** [comment]
added return values

**Comment by SauravDharwadkar** [comment]
and check things again new in github and stuff

File: CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (ed-asriyan) [reply CMakeLists.txt:19]: okay

File: main.cpp [hunk 2]
```diff
@@ -58,19 +74,28 @@ int main(int argc, const char** argv) {
...
+        std::cerr << "Error in is_directory: " << ec.message();
+	return 1;
+    	}
+    if (fs::is_regular_file(path, ec)){
+		return convert(filePath,width,height,alphaThreshold,output,fps);
+		}
+    if (ec){
+        std::cerr << "Error in is_regular_file: " << ec.message();
+	return 1;
+    }
```
[the remark this answers is not in the document]
Comment (SauravDharwadkar) [reply main.cpp:98]: spaces and tabs , fixed

[the remark this answers is not in the document]
Comment (SauravDharwadkar) [reply main.cpp:98]: https://github.com/SauravDharwadkar/tgs-to-gif/blob/f207095597ddf7e46a12de340dff4caaa1ed5c67/main.cpp it was hidden

**Comment by SauravDharwadkar** [comment]
Let me start my pc 😅

File: main.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (SauravDharwadkar) [reply main.cpp]: in editor i didnt see intend sorry to bother ed

**Pull request merged**